Formula/sorbet.rb

```diff
@@ -0,0 +1,22 @@
+class Sorbet < Formula
+  desc "Fast, powerful type checker designed for Ruby"
+  homepage "https://sorbet.org"
+  url "https://github.com/sorbet/sorbet/archive/0.4.5059.20191122150919-e82c71a19.tar.gz"
+  version "0.4.0"
+  sha256 "6f7921c4689ced005f2ee5cf82c6470a876f3b70d32161e6dacbb4bf2fa10c32"
+
+  depends_on "bazel" => :build
+  depends_on "llvm" => :build
+
+  def install
+    # jemalloc is not set because it does not build on all platforms
+    # specifically on Mojave since it builds with Bazel, it does not get the correct header paths for the SDK.
+    # the headers are searched only /usr/include which have been removed in Mojave
+    system "bazel", "build", "--define=release=true", "--compilation_mode=opt", "--config=debugsymbols", "--config=static-libs", "--config=versioned", "//main:sorbet"
+    bin.install "bazel-bin/main/sorbet" => "sorbet"
+  end
+
+  test do
+    assert_match /#{version}/, shell_output("#{bin}/sorbet --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Added better test + fixed version